File: test/unit/code_parser/whole_tests/sources/dc_2015.rb
with_dc do |dc|
  comment "Settings specific to other formats created with from PDF conversions"
  comment "are set using the FromPDFOptions object"
  createobject "fPDF", :com => "APDocConverter.FromPDFOptions", :class => "APDocConverter.FromPDFOptions", :call => "APDocConverter.FromPDFOptions()", :path => :"C:\\Program Files\\ActivePDF\\DocConverter\\bin\\APDocConverter.Net35.dll" do
    br
    comment "To Word options"
    if @language_type == :net
      ToWordHeadersAndFootersMode :equals => :"APDocConverter.ToWordHeadersAndFootersOptions.Detect"
    else
      ToWordHeadersAndFootersMode :equals => 0
    end
    br
    comment "To Excel options"
    if @language_type == :net
      ToExcelAutoDetectSeparators :equals => true
      ToExcelTablesFromContent :equals => :"APDocConverter.ToExcelTablesFromContentOptions.Default"
    else
      ToExcelAutoDetectSeparators :equals => true
      ToExcelTablesFromContent :equals => 0
    end
    br
    comment "To Image options"
    if @language_type == :net
      ToImagePageDPI :equals => 300
    else
      ToImagePageDPI :equals => 300
    end
    br
    comment "Send the from PDF settings to DocConverter"
    dc.SetFromPDFOptions @object_variable.to_sym
  end
  br
  @file_doc = "PDF.docx"
  @file_pdf = "PDF.pdf"  
  comment "Convert the document from PDF to another format"
  comment "The second parameter determines the output file format"
  if @language_type == :net
    case @language
    when "cf"
      output_for_language :cf => 'dcFromPDF = CreateObject(".NET", "APDocConverter.FromPDFFunction", "%s");' % [@vars[:product][:path]]
      @from_pdf_value = :"dcFromPDF.ToWordDocX"
    else
      @from_pdf_value = :"APDocConverter.FromPDFFunction.ToWordDocX"
    end
  else
    comment " 0 = .doc"
    comment " 1 = .docx"
    comment " 2 = .xls"
    comment " 3 = .ppt"
    comment " 4 = .html"
    comment " 5 = .txt"
    comment " 6 = .bmp"
    comment " 7 = .jpg"
    comment " 8 = .png"
    comment " 9 = .tif"
    comment "10 = .pdf (PDF/A)"
    comment "11 = Verify PDF/A compliance"
    comment "12 = .rtf"
    comment "13 = .gif"
    comment "14 = .tif (multipage)"
    @from_pdf_value = 1
  end
  ConvertFromPDF relative_file(@file_pdf), @from_pdf_value, relative_file(@file_doc), :assert_result => "DocConverter"
end

File: test/unit/code_parser/whole_tests/views/dc_2015.php
<?php
// Copyright (c) 2000 ActivePDF, Inc.
// ActivePDF DocConverter 2015
// Example generated 01/01/2000 
?>

<?php
// Get current path
$strPath = dirname(__FILE__) . "\\";

// Instantiate Object
$oDC = new COM("APDocConverter.Object");

// Settings specific to other formats created with from PDF conversions
// are set using the FromPDFOptions object
$fPDF = new COM("APDocConverter.FromPDFOptions");

// To Word options
$fPDF->ToWordHeadersAndFootersMode = 0;

// To Excel options
$fPDF->ToExcelAutoDetectSeparators = true;
$fPDF->ToExcelTablesFromContent = 0;

// To Image options
$fPDF->ToImagePageDPI = 300;

// Send the from PDF settings to DocConverter
$oDC->SetFromPDFOptions($fPDF);

// Release Object
$fPDF = null;

// Convert the document from PDF to another format
// The second parameter determines the output file format
//  0 = .doc
//  1 = .docx
//  2 = .xls
//  3 = .ppt
//  4 = .html
//  5 = .txt
//  6 = .bmp
//  7 = .jpg
//  8 = .png
//  9 = .tif
// 10 = .pdf (PDF/A)
// 11 = Verify PDF/A compliance
// 12 = .rtf
// 13 = .gif
// 14 = .tif (multipage)
$results = $oDC->ConvertFromPDF($strPath . "PDF.pdf", 1, $strPath . "PDF.docx");
if ($results->DocConverterStatus != 0) {
  Error("ConvertFromPDF", $results, $results->DocConverterStatus);
}

// Release Object
$oDC = null;

// Process Complete
echo "Done!";

// Error Handling
function Error($method, $oResults, $errorStatus) {
  echo "Error with " . $method . ": <br/>"
    . $errorStatus . "<br/>"
    . $oResults->details;
  exit(1);
}
?>